Zend/tests/enum/traits-no-properties.phpt

```diff
@@ -0,0 +1,25 @@
+--TEST--
+Enum cannot have properties, even via traits
+--FILE--
+<?php
+
+trait Rectangle {
+    protected string $shape = "Rectangle";
+
+    public function shape(): string {
+        return $this->shape;
+    }
+}
+
+enum Suit {
+    use Rectangle;
+
+    case Hearts;
+    case Diamonds;
+    case Clubs;
+    case Spades;
+}
+
+?>
+--EXPECTF--
+Fatal error: Enum "Suit" may not include member variables in %s on line %d
```

Comment:
I think it would be good to also check the declaration of a forbidden method, such as `__construct` and `cases` in the trait.

Reply:
Hm, traits don't throw when methods are already declared in the given class. So I guess the consistent behavior is to ignore the method from the trait and use the cases method declared in the enum under the hood?